### PR TITLE
196 close websocket on the backend

### PR DESCRIPTION
**Pull request by merituulie**
Fixed the AI opponent & token flow, close the websocket and return after the game is over.

### CODE REVIEW

File: backend/Match/views.py [hunk 1]
```diff
@@ -15,7 +15,7 @@ class MatchView(APIView):
     @method_decorator(must_be_authenticated)
     def get(self, request):
         token_str = request.query_params.get('token')
-        if not token_str:
+        if not token_str or token_str == 'null':
```
Comment (dorianjagusch) [inline backend/Match/views.py:18]: Does python treat the null as true?

File: backend/pong/consumers.py [hunk 2]
```diff
@@ -47,16 +45,18 @@ async def connect(self):
...
+        await self.close()
 
     async def receive(self, text_data):
         key_press = text_data.strip()
         self.game.update_positions(key_press)
 
         if self.game.game_stats.game_over == True:
             await self.send_positions()
-            await self.save_match_results()
+            await self.save_match_final_results()
```
Comment (dorianjagusch) [inline backend/pong/consumers.py:57]: So this is supposed to be back to final result?

**Review by dorianjagusch — APPROVED** [review]

File: backend/pong/consumers.py (hunk 2 above)
Comment (sakarisalmi) [reply backend/pong/consumers.py:57]: There's some git shenanigans going on.

File: backend/pong/consumers.py [hunk 3]
```diff
@@ -1,19 +1,17 @@
...
 import json
 import asyncio
+from asgiref.sync import sync_to_async
 from channels.generic.websocket import AsyncWebsocketConsumer
 from .game import PongStatus
 from .constants import *
 from channels.db import database_sync_to_async
 from Tokens.models import MatchToken
 from Match.models import Match
-from Match.matchState import MatchState
```
Comment (sakarisalmi) [inline backend/pong/consumers.py:11]: The imports are necessary. I recommend copy-pasting the stuff you've added in this pr to a pulled file from main. Some of the changes in this file tells me that something went wrong with git or something of the sorts.

File: backend/Match/views.py (hunk 1 above)
Comment (merituulie) [reply backend/Match/views.py:18]: nope, this is a hack, the frontend given it does not find the token will send a string 'null' to the back

File: backend/pong/consumers.py (hunk 3 above)
Comment (merituulie) [reply backend/pong/consumers.py:11]: There is some weirdness with the commits to this file for sure.... has there been a sqash merge or something?

File: backend/Match/views.py (hunk 1 above)
Comment (dorianjagusch) [reply backend/Match/views.py:18]: ohhhh

File: backend/pong/consumers.py (hunk 3 above)
Comment (dorianjagusch) [reply backend/pong/consumers.py]: Still shouldn't be possible at least not squashing PRs onto the main. What people do on their branches is up to them (so maybe?)

**Review by dorianjagusch — APPROVED** [review]

**Review by sakarisalmi — APPROVED** [review]
Nice, everything looks good!

**Pull request merged**